### ISSUE TITLE
Decouple CLI install From inject

### ISSUE THREAD

**Issue by ihcsim** [linkerd/linkerd2#2385]
The current CLI `install` code depends on the `inject` code to modified the Kubernetes resource YAML manifests. This coupling requires us to modify the `inject` code  to depend on the `globalConfig` and `proxyConfig` pb structs even though the `inject` code doesn't require it. Once #2334 is merged, we should be able to break this dependency, and update `install` to call the `/pkg/inject/` lib.

**Comment by grampelberg** [comment]
To get to the point where a helm chart is published, the `install` flow will need to do inject somehow outside of the usual inject flow (or maybe they need to be unified?). This is obviously not a blocker or anything, just something to think about while the code's being moved around.

**Comment by olix0r** [comment]
@ihcsim Is this issue still relevant given the recent inject changes?

**Comment by ihcsim** [comment]
Closing this issue, as it's no longer relevant.
